### PR TITLE
♻️ Maintenance: Fixes and removing deprecated functionality in `servicelib`

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

Big cleanup of deprecated/unused code in `servicelib`

- 🗑️ Removing dependencies to `attr` (replaced by `dataclass`) 
- 🗑️ Removing deprecated, duplicated or unused functionality
   - submodules as `servicelib.aiohttp.rest_*` ...  which are related `openapi-core`
   - functionality resources 
- 🐛 Fixes bugs detected by mypy
- ⚗️  SEE `test_application.py` app events
      -  found some pitfalls on cleanup of client and fire&forget events!

## Related issue/s

- Implementing https://github.com/ITISFoundation/osparc-simcore/issues/2008

## How to test

- in-place

## DevOps

None

### CODE REVIEW

**Review by matusdrobuliak66 — APPROVED** [review]
looking good :+1:

File: packages/service-library/src/servicelib/aiohttp/application.py [hunk 1]
```diff
@@ -1,60 +1,82 @@
...
+async def _first_call_on_cleanup(app: web.Application):
+    # NOTE: name used in tests to mocker.spy
+    _logger.info("Shutdown completed. Cleaning up %s ...", f"{app}")
 
 
-async def stop_background_tasks(app: web.Application):
-    running_tasks = app[APP_FIRE_AND_FORGET_TASKS_KEY]
-    task: asyncio.Task
-    for task in running_tasks:
+_MAX_WAIT_TIME_TO_CANCEL_SECONDS = 5
```
Comment (GitHK) [inline packages/service-library/src/servicelib/aiohttp/application.py:22]: We are using this naming convention elsewere 
```suggestion
_MAX_WAIT_TIME_TO_CANCEL_S: Final[int] = 5
```

File: packages/service-library/src/servicelib/aiohttp/application_setup.py [hunk 2]
```diff
@@ -62,12 +63,13 @@ class SetupMetadataDict(TypedDict):
 
 def _parse_and_validate_arguments(
     module_name, depends, config_section, config_enabled
-) -> Tuple:
+) -> tuple:
```
Comment (GitHK) [inline packages/service-library/src/servicelib/aiohttp/application_setup.py:66]: could you specify the types returned in this tuple as well?

File: packages/service-library/src/servicelib/aiohttp/openapi_wrappers.py [hunk 3]
```diff
@@ -113,7 +115,7 @@ def __init__(self, response: web.Response, data: str):
 
     @staticmethod
     async def create(response: web.Response):
-        text = await response.text()
+        text = await response.text
```
Comment (GitHK) [inline packages/service-library/src/servicelib/aiohttp/openapi_wrappers.py:118]: are you sure about this one? I think aiohttp has a function text(). Would you mind double checking it?

File: packages/service-library/src/servicelib/utils.py [hunk 4]
```diff
@@ -113,8 +112,10 @@ async def sem_task(task: Awaitable[Any]) -> Any:
                 return await task
 
         wrapped_tasks = [sem_task(t) for t in tasks]
+    else:
+        wrapped_tasks = tasks
```
Comment (GitHK) [inline packages/service-library/src/servicelib/utils.py:116]: what is the advantage in moving this here? I would have done it as it was before.

**Review by GitHK — APPROVED** [review]
Nice, find some questions below.

**Comment by mrnicegyu11** [comment]
thanks!

File: packages/service-library/src/servicelib/utils.py (hunk 4 above)
Comment (pcrespov) [reply packages/service-library/src/servicelib/utils.py:116]: linter suggested that

File: packages/service-library/src/servicelib/aiohttp/openapi_wrappers.py (hunk 3 above)
Comment (pcrespov) [reply packages/service-library/src/servicelib/aiohttp/openapi_wrappers.py:118]: https://docs.aiohttp.org/en/stable/web_reference.html#aiohttp.web.Response.text

File: packages/service-library/src/servicelib/aiohttp/application_setup.py (hunk 2 above)
Comment (pcrespov) [reply packages/service-library/src/servicelib/aiohttp/application_setup.py:66]: but they are the same as the input arguments ... :-)

File: packages/service-library/src/servicelib/aiohttp/application.py (hunk 1 above)
Comment (pcrespov) [reply packages/service-library/src/servicelib/aiohttp/application.py:22]: I think `SECONDS` is more clear than `S`. Moreover is a protected variable  :-)

**Pull request merged**